src/Tools/ExternalAccess/LiveShare/Tagger/ModeAwareSyntacticClassificationTaggerProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Experiments;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.LiveShare.Tagger
+{
+    /// <summary>
+    /// this is almost straight copy from typescript for syntatic LSP experiement.
+    /// we won't attemp to change code to follow Roslyn style until we have result of the experiement
+    /// </summary>
+    [Export(typeof(ITaggerProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    [TagType(typeof(IClassificationTag))]
+    [ContentType(StringConstants.CSharpLspContentTypeName)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    internal sealed class CSharpModeAwareSyntacticClassificationTaggerProvider : ModeAwareSyntacticClassificationTaggerProvider
```

Comment:
what does "ModeAware" mean?

Reply:
it means it aware textmate or syntaxoverlsp mode aware.